### DISCUSSION TITLE
Maybe we could find a way to reduce extreme runs of luck

### DISCUSSION THREAD

**Discussion by peyre** [raceintospace/raceintospace#542] [Ideas]
This came to mind again yesterday.  My very first manned Vostok mission suffered a catastrophic failure that reduced it to 6%.  At that point there seemed no point in bringing it back up to speed - it would've been too little, too late.  So I put everything into Lapot since it couldn't benefit from tech transfer anyway, and on my second manned mission it too killed everyone aboard and dropped it all the way back to 6%.  So frustrating!  And I had done everything right, too, bringing the hardware up to Max R&D and even flying a couple dummy tests.

I realize this is what happens in a random distribution - sometimes you just get a run of very good or bad luck, and random rolls are an integral part of the game mechanics.  Still, I'm reminded that most of the catastrophic failures in the Space Race occurred when the agency did something <s>stupid</s> --er, incautious: running a sim of the Apollo capsule with pure oxygen at one atmosphere, pushing ahead with the first Soyuz flight even when the engineers shouted it wasn't ready, etc.  

The way the game works, it's nearly as likely to wipe you out whether you're taking all the precautions or cutting every corner.  One of the early complaints about BARIS was that it gave you the feeling you were being punished for getting bad rolls and not because of anything you did wrong.  I just wonder if there's something we can change in the game mechanics to go easy on you if you've flown dummy tests or if you've suffered one catastrophic failure already (within, say, 6 turns or so).  There could also be a flip side to it, where you're likely to suffer if you're flying people without even getting things up to Max R&D.

**Comment by smudog** [comment]
I'm not sure if its in the mechanics already - but what about getting a free failure pass for a crewed mission if you have a proceeding catastrophic failure on an uncrewed mission.  That always bugged me that there seemed to be no reward for blowing up a test mission - that is typically the most educational part of an given program program!  Of course, there would be the perverse incentive to fly uncrewed before you had done your R&D - but then again - I think that is exactly what is happing at Boca Chica right now....

**Reply by peyre** [reply]
That's an intriguing thought, giving an Avoid Fail card on a failed dummy test.  We could get around your concern with it by giving the card only if the item is at or above Max R&D - that should prevent abuse of the system.

What I was thinking in the OP was finding some way of protecting a player who's done everything right from suffering two bad catastrophic failures back to back, especially when, not only have they done their due diligence in regard to safety, but they haven't derived any benefit from the vehicles that keep failing.  

An Avoid Fail card might be helpful with manned flights as well - though oftentimes when I see this, it strikes my Mercury/Vostok, which I then abandon for a more advanced program.  Also, perhaps it shouldn't get the AF card until it reaches / returns to Max R&D, again to prevent jumping ahead for a prestige boost using a new spacecraft that's not really ready yet.

**Reply by peyre** [reply]
Another possibility (which might be a little trickier to implement) would be that a component that suffered a catastrophic failure would receive an Avoid Fail card once it's at Max R&D _and_ has flown one dummy test (Unmanned Suborbital or Orbital).

**Reply by peyre** [reply]
I just had another of these games where I just. could. not. get. ahead.  I hadn't even flown a second unmanned docking mission or launched a lunar flyby, when the AI sailed over to the Moon in 1969.  I think this idea of giving an avoid fail card (once the affected hardware has returned to Max R&D) is a really good one.

For an in-game justification, we could consider that after a disaster, your people will be extra careful and thorough in all their procedures - provided you're taking time to do things right (which is established/assumed by the fact that you've researched it as much as you can).

**Comment by peyre** [comment]
Here's an idea: when one side is having too much good fortune, maybe rather than suffering forced catastrophic failures they could receive negative newscasts.  Conversely, a player who's doing poorly through no fault of their own could perhaps receive positive newscasts.

**Reply by rnyoakum** [reply]
Newscasts are assigned at the beginning of the game, so this would require some major work on the newscast system. It seems a little overkill.

> when one side is having too much good fortune, maybe rather than suffering forced catastrophic failures they could receive negative newscasts

No. Just no. You're stuck trying to decide what is "good fortune" vs good strategy, and that's not something easily defined in an algorithm. It's also the bold move of deciding that someone is having _too much_ fun now, so they should suffer a bit, when you've no idea what the Random Number Gods have planned.

**Reply by peyre** [reply]
Ah.  Well it seemed like a good idea at the time.

**Reply by peyre** [reply]
I see that implementing _this_ idea would entail setting the variable for avoiding newscasts, then inserting a check into news_suq.cpp for that variable and rerolling certain `case`s, then clearing or reducing the variable.  Tricky.  I think the avoid fail idea is probably a better one.

**Comment by hweimer** [comment]
Well, not all catastrophic failures were because of stupidity, just think of Soyuz 11 or STS-107. There were also a couple of close calls that were essentially random events (Gemini 8, for example). What one could do, would be to increase the chance of a catastrophic failure the more the die roll surpasses the safety factor of the step involved. But that would affect the balance of the game by quite a lot and make the endgame much less thrilling.

**Reply by peyre** [reply]
Good points there.  Say, what if dummy tests run with a piece of hardware reduced the risk of catastrophic failure when you have a failed roll with that component?  Capsules/shuttles anyway, probably not rockets.

**Reply by hweimer** [reply]
Well, the risk is already reduced because the dummy test awards +1% in safety. Reducing it further does not seem to be justified to me. Spaceflight _is_ pretty dangerous at the end of the day.

**Reply by peyre** [reply]
Quite true.  I was just thinking that, in the event of a failure, the chances of recovery might be better if you've taken the time to run dummy tests.  As it is, the game doesn't give you much reason to fly them...though of course, Unmanned Docking Tests double as dummy tests under the current system, so would they count as well?  I'm inclined to think they shouldn't, that only Unmanned Suborbital, Unmanned Orbital, Unmanned Lunar Pass (and Orbital) should count if we did something like this, and then only once apiece, maybe.

**Reply by hweimer** [reply]
>  I was just thinking that, in the event of a failure, the chances of recovery might be better if you've taken the time to run dummy tests.

What one could do is that on level 1, if a component catastrophically fails on its very first mission, then you will be thrown back to base safety (like on level 2 & 3) instead of the safety factor being cut in half.

> As it is, the game doesn't give you much reason to fly them...

Hmm, I usually fly several dummy test in a game, even some which are not unmanned docking.

**Reply by peyre** [reply]
Oh, that's a good point I hadn't thought of.  I had another idea: maybe flying dummy missions before the cat failure would raise the Max R&D it drops to (so Soyuz would drop to 86% instead of 85% if you had flown one dummy test).  But that would be complex to implement, and on second thought it would probably unfairly advantage Gemini, which we don't need.  Oh well.

Really!  Wow, it's funny how playing styles differ.  I usually don't fly more than one straight-up dummy test (ie, one that isn't also docking) per capsule.

**Reply by hweimer** [reply]
One way to make dummy tests more realistic is to change the reward structure. Maybe something like the first successful use of a component gives you 3 points, the second 2, and any subsequent one has only a 50% chance to award 1 point. This would make early tests much more valuable, while at the same time addressing the making-gemini-unbeatable-by-15-unmanned-docking-missions thing.

**Reply by peyre** [reply]
That's another option.  Or how about if U. Suborbital and U. Orbital gave 2 points per flight?  If we were to do that I'd also suggest U. Lunar Pass and Orbital do the same for the kicker that's used.  That would make them really useful for Gemini/Voskhod and minishuttles.

**Reply by hweimer** [reply]
I think it would be somewhat strange if U. Orbital gave you two points, but U. Orbital Docking gave you only one. After all, one is a subset of the other. I'm also not so sure if having something particularly useful for Gemini/Voskhod would help with balancing the game, but maybe that's just me. Personally, I think Gemini/Voskhod is borderline on being too powerful, on the other hand, Lunar Gemini would have beaten Apollo by quite a bit in real life.

**Reply by peyre** [reply]
Ok, well as usual, I'm casting out ideas.  I think of good ones and bad ones.  I certainly don't want to give additional advantages to Gemini.

My thinking with the 2 points for those missions are that they're essentially _dedicated_ dummy missions.  The point that they could effectively halve the time a Gemini/Voskhod needs to get back up to speed hadn't occurred to me.

**Reply by sflynn850** [reply]
Do we have a spreadsheet or anything that illustrates cost+risk to reward ratios of various platforms and missions? That might help us decide how to balance some of these things.

**Reply by hweimer** [reply]
Leon, I totally appreciate your ideas. I think it is really hard to find solutions that are a) balanced, b) fun, and c) historically realistic, all at the same time. Nudging people towards dummy tests certainly helps with c), although we will always have less missions than there were in real life.

I once started such a spreadsheet, but then gave up at some point, because you would have to include large parts of the mission logic (e.g., how likely it is for a failed roll on a given step to be catastrophic or lead to the mission being scrubbed). I also made some quick calculations on how much money it costs to get the entire equipment stack to 90%. Perhaps not too surprising, Gemini came out on top, but this didn't include things like the additional complexity of doing a joint lunar landing mission. And there are some things that are extremely hard to evaluate in such a setting. For instance, EOR or LOR? Sure, EOR has less mission steps, so less chance for failures, but LOR actually has one *manned* mission step less than EOR (no earth orbital activities) and also doesn't require boosters on the manned launch (thanks to the lighter kicker), so less chance for a *catastrophic* failure.

**Reply by peyre** [reply]
Thanks.  I'm pretty good at thinking of things that might help with b) fun or c) historically realistic, but most of them turn out to conflict with a) balanced, or even with each other (b & c).

For my money, the extra manned mission step on EOR is a lesser issue than the several extra unmanned steps with LOR.  Sure there's no chance for catastrophic failure on those steps, but on the flip side there's a much greater chance of something going wrong on one of them, than on the single extra manned EOR step, and a failure on an unmanned step always fails the mission.  I suppose for purposes of your calculations there you could consider EOR vs. LOR a wash.

**Reply by hweimer** [reply]
Okay, now I got really curious to look at the numbers. LOR has two more unmanned steps (TLI and lunar orbit ins burn), so that's close to 20% increase in failure probability at 90% safety. A failure will delay the lunar landing by one turn (assuming that you have scheduled a second attempt), i.e., a delay of 0.2 turns on average. EOR's additional earth orbital activities step has about 20% chance of being catastrophic, and 40% chance of scrubbing the mission. Again, at 90% safety, that's an average delay of 0.1 * 0.4 + 4 * 0.1 * 0.2 = 0.12 turns, if we assume that bringing Gemini back to 90% safety takes 4 turns for R&D and dummy tests. So, yeah, you are right that EOR is the better one.

**Comment by peyre** [comment]
A related idea might be for U. Suborbital and U. Orbital to always raise the capsule's Safety and Max R&D by a point even if it doesn't succeed on any steps of the mission.  That would encourage flying dummy missions...though it might also invite abuse by someone flying it at, say, 23%.  

The rationale would be that that mission was chosen specifically to iron out kinks in the hardware.

Edit: possibly, the Max R&D going up could apply only if the capsule is at Max R&D already, or perhaps at 75% (the top threshold for gaining technology transfer).

**Comment by peyre** [comment]
I just had another of these!  After my first suborbital I just couldn't make any headway!  Everything failed.  Castastrophic failure followed by newscasts repeatedly slapping down the Safety of components I was already having trouble researching because my budget sucked.  I couldn't even fly a Duration B until Spring 1969 and never got my Ranger up to speed.  When the computer swept in with the Moon landing in Fall '69 it was kind of a relief to be done with that game.

**Comment by peyre** [comment]
Having yet another of these "piece of sh**" games where it just keeps dumping on me.  The AI's at level 3, yet it can do no wrong and obstructions keep being thrown my way.  The AI flew an orbital before I could even get a satellite up, it flew a manned lunar pass back in '65 or '66, and in Spring '67 it looks to be a step or two away from the landing.  Meanwhile the newscast (seemingly "strategically", if I believed that sort of thing) damages my equipment and then when I'm back up to speed it points and laughs, "HA ha, no rockets for a year! Sucks be to you!"

It really would be helpful if there were a way to have the game avoid negative newscasts when there's a huge discrepancy in game score or something.

![Disproportionate](https://user-images.githubusercontent.com/3118014/145723972-72cad865-b5c3-44c7-a4aa-6a5def1d3394.png)
![NoRockets](https://user-images.githubusercontent.com/3118014/145724103-2d61862e-871f-4a0e-9350-f415ec9bca12.png)

Edit: sure enough, the AI cruised to an easy victory in Spring 1968, before I had even done a lunar flyby or Duration D.
![AIWin](https://user-images.githubusercontent.com/3118014/145726317-c289b87c-a4e4-434e-832e-323055c89ecc.png)

**Reply by rnyoakum** [reply]
Newscasts are assigned at the beginning of the game, and while some event cards are restricted to certain years, they're otherwise random. Best I can tell, there's no attempt to balance things, other than to prevent repetition of events. That means some games are not going to be in your favor, which reflects the random nature of the board game. That's not restricted to the proportions of good/bad/neutral events; early events have a much greater impact than events in the late 70s. Also, event cards can't recur, for you OR your opponent, so if you're getting bad events your opponent may statistically benefit from your misfortune.

But Race Into Space can be anything it wants to be. Here's a hypothetical approach that would 'balance' things:
- News events are chosen era by era.
- In each era, all the available event cards (those not chosen) are collected together. 'Expired' events restricted to previous eras are removed from the pool, and the newly available events for that era are added to the pool.
- The card pool is sorted into 'good', 'bad', and 'neutral' piles.
- each player draws a certain number of cards from each pile, proportion determined by difficult level, to represent the events of that era. These cards are shuffled to create a random order of events for that section of the game.

EDIT: Integral to the event system I described is being balanced for both players, though potentially biased by difficulty level. In a game with two human players, that's important; it wouldn't be fair to perform a sleight-of-hand on the newscasts if one player's advantage is owed to superior strategy. For a game against an AI, blessed subterfuge is a legitimate option.

**Reply by peyre** [reply]
That sounds like a reasonable approach!  Most of the time, the game is reasonably balanced, but there are times when one side constantly rolls snake eyes, and it would be good if we could moderate some of the excesses.

**Reply by peyre** [reply]
I had been thinking about some kind of system that would prevent (or make less likely) receiving 2+ negative newscasts in a row, but your system there would largely take care of that.

**Comment by peyre** [comment]
Another idea that's come to mind is, maybe the game should dock your budget less if the component that failed was at least Max R&D, as a sort of recognition that you at least weren't being reckless with it.  But...that might sort of contradict the whole point of the prestige system...

**Comment by hweimer** [comment]
A quite different approach would be to have (un)lucky breaks not to have such a large impact on the game to begin with. Right now, the game rewards you if you are successful and penalizes you if you are not. I am not sure that this is always a realistic model. For example, following the Sputnik shock, the Western world has invested an incredible amount of resources into science and technology, i.e., quite the opposite of what happens in-game right now. Similarly, I recently had a game where I got a spring half-off sale with something like 170 MB cash on hand, which allowed me to land on the moon in the fall of 1966. I can't imagine a real-world congress saying "yeah, we said we would go to the moon in 1969, but we have no problem of giving you even more money to get there a few years earlier and make the president look like a moron that didn't know how long this would take".

Maybe one could extend the VP review to create a government inquiry after each catastrophic failure. If you didn't take any excessive risks (not flying at max R&D, no dummy tests, etc.), you are cleared of any wrongdoing and your budget gets restored.

**Reply by peyre** [reply]
That sounds like a very complicated solution to implement, but also very fair.  It would certainly help in games where you're suffering from a string of setbacks!  Those are extremely discouraging.

**Reply by Angel-0a** [reply]
> quite the opposite of what happens in-game right now.

I just thought about the same thing. The USA applied serious resources to the space program after two spectacular prestige failures, not successes. Maybe the game could throw some extra money at a losing player every turn based simply on a difference in prestige? This was a race after all and this would reflect nation's determination to catch up.

**Reply by hweimer** [reply]
I think the big question is whether the difference in prestige is due to random events or difference in skill. We certainly don't want to reward a player who suffers from bad prestige after some stupid choices. And how are we supposed to judge this? Sometimes, skillful play requires you to cut corners, which in other circumstances would be foolish.

**Reply by Angel-0a** [reply]
> We certainly don't want to reward a player who suffers from bad prestige after some stupid choices.

I don't think this would be that much of a problem because I don't think 20 or 30 MBs per fiscal year could save a bad player anyway (but could help a struggling good player to recover). I would be more concerned about a possibility of turning this into an exploit, like tanking your prestige on purpose with some cheap underdeveloped hardware while developing your main hardware with extra money. On the other hand I imagine this bonus system would work on top of the current one, which keeps the losing player poor, so simply winning would still be the better strategy.

**Comment by peyre** [comment]
My son had a good idea this morning.  If we were to find a way to smooth the "lumpiness" of runs of luck, maybe that smoothing could be lessened at difficulty level 2 and not done at all at level 3.

**Comment by hweimer** [comment]
Right now, I am playtesting PBEM (more on that later), so I play against myself using virtually identical strategies. What I've noticed, is that the course of the game strongly depends on who comes out first during the Mercury/Vostok age. Once the low-hanging fruits are taken, it is basically impossible for the player who has been on the short end to buy and research Titan/Proton for the lunar flyby or the two-seater because the budget is too low.

So, I'm inclined to think that it's not runs of luck that are the problem, but rather luck at a pivotal point in the game. In games against the AI, this is less noticeable because the AI can catch up by cheating, and the human player can catch up by playing better.

**Reply by peyre** [reply]
That's an important discovery!  Maybe there's something we can do to compensate for the resulting disparity.  Maybe, the other player gets a one-time 10MB grant from the government so we can catch up to those da*ned commies, or whatever?  Or maybe the use of the one-man capsule enabled the CIA to find intelligence on it, thereby triggering a newscast giving your Mercury x% Safety.

I have to say though that I have quite a number of games where the AI beats me to first in orbit, but I catch up later.  I don't know what percentage that is though.

The extreme runs of luck I'm thinking of are where you have a catastrophic failure, and as soon as you get back up to Max R&D territory, you have another.  Then when you're about to recover from that, you get a "no rockets for a year".  So a _series_ of slap-downs in a row.  That's what I find most infuriating in this game, when it occurs.

How do we enable PBEM, by the way?

**Reply by rnyoakum** [reply]
The 'pivotal moment' theory brings me back to the idea I'd thrown out last year about generating news events based on eras. I'd been thinking about delaying event assignment until certain years - corresponding roughly with game play cycles - but a milestone approach might work better.

In this model, the set of news events is initially generated for both players (as currently happens). Potentially, those events could be nudged from **random** to **balanced** based on player difficulty. At given milestones, the game stops and 'reshuffles' the news events. Previously encountered cards are obviously untouched, but the News cards for the remaining turns are shuffled back into the deck and re-dealt. Accounting for player difficulty and comparative achievements, these events would be **random** or **balanced for/against** the player.

This approach compensates for early runs of luck behind the scenes rather than overt action. Also, it would stop the RNG from kicking a player while they're down as badly (freak mission accidents would still be a danger, but random dice rolls _are_ inextricably the underlying mechanic for the game). Obvious downsides? Meddling with the news events for balance or benefit would probably be tricky, and would need to be tinkered with to hit the sweet spot. Not to mention, any sort of interference in the flow of the game has its own concerns. Players take risks in the hope of rewards. Bailing out a player whose risks don't pan out takes away the thrill _and_ encourages bad risk-taking. In PvC, who cares? Use the player difficulty level to determine how they want to be treated. But in PvP, a "risk meter" might be needed to avoid helping players who rely on daring too much. That would mean having to build a "risk meter."

Thoughts?

**Reply by peyre** [reply]
That sounds like one of my suggestions, only much better thought out.  One problem with using player difficulty (if I'm not mistaken) is that level 2 adds a 3% penalty to every step of every mission, and level 3 adds 6%.  So playing above level 1 is already pretty harsh; I'm not sure it would make a very effective means of determining whether the player wants the newscasts to go easy on them ("I'd like it to not go easy on me, but I don't want that BS 3% penalty on everything" for instance).

That's a very valid concern, that we could drag this too far in the "easy" direction to where we encourage risky behavior.  A risk meter is a great idea, though working out the mechanics of it might be tricky.

Maybe newscasts could be rated on a -2 to 2 scale, where 0 is a newscast that doesn't affect your space program, 1 or -1 is somewhat helpful/hurtful (such as getting the 3 Air Force pilots), and 2 or -2 is heavily beneficial/hurtful (like half-off sales or no rockets/capsules for a year).  Newscasts could be prevented from doing -2 events for a year after a catastrophic failure.  (Hm.  This mechanic seems more sketchy the more I flesh it out.)

**Reply by hweimer** [reply]
I think both of you raise some very good points. The more I think about it, the more I become convinced that what we really want is to have the random nature of the game to affect both players in a correlated way. So, if you have a series of bad newscasts over a given era, the other player would be similarly affected. However, the marginals of the distribution would be unchanged, so seeing a set of bad news would not give you any information about future events. This could even be extended to rolls against mission safety factors, so you could ensure that both players get the same amount of rolls, say larger than 75, 85, and 95, for example.

I have some experience in dealing with correlated random numbers, so I'll have a look what I can come up with.

**Reply by hweimer** [reply]
Addendum: I think dealing with setbacks is an important part of the game and an area where skill can make an important difference. Therefore, I think we should not reduce the amount of setbacks, but rather make sure that they are more or less evenly dealt to both players.

**Reply by peyre** [reply]
Ryan, maybe your idea could prevent things like this.  Last turn I was _just about_ to launch my minishuttle for the first time, ready to catch some of that sweet, sweet prestige after my one-man capsule burned up on reentry, when:

![NewscastDowngrade1](https://user-images.githubusercontent.com/3118014/193421087-7d163435-5279-47cb-896a-61f11cc8daf5.png)

Ok fine, so I'll research it and wait a turn.  Then what do I see?
![NewscastDowngrade2](https://user-images.githubusercontent.com/3118014/193421086-4bcea034-2af3-40f4-b191-2bc43f912aa9.png)

It's occasions like this that give you the feeling "Why even bother?"

Especially when it's followed by failure after failure - first the shuttle fails to get off the ground, then it fails to achieve an orbit, and then on the third try, it explodes.  It's 1965 and I haven't even completed an orbital, and the computer has landed a probe on the Moon while I haven't been able to launch a flyby attempt yet. (The Americans flew a lunar pass the following turn and a lunar orbit the next, and landed in Spring 1967 with **Apollo I** before I could launch another shuttle.)
![Explodes](https://user-images.githubusercontent.com/3118014/193421482-90d2c93c-254f-45f9-b464-ed2f147b49b8.png)

Hello boys!  (And this is _before_ the lunar pass and orbital.)
![RankingDisparity](https://user-images.githubusercontent.com/3118014/193421556-92cf1ea2-2140-457f-94b7-cc0dd613acdc.png)

Incidentally, this one is a good illustration of the disparity Ryan mentioned - that being the first to achieve orbit puts you way ahead.  Here the computer did suborbital, orbital, and EVA all in one mission, as I recall.

**Reply by peyre** [reply]
Hendrik, I agree entirely.  Setbacks are an integral part of the game and I wouldn't want to eliminate them.  I only object to these kinds of cascade failures like I just experienced.

**Reply by peyre** [reply]
Ryan, here's another visual that might help show what you're talking about above.  In this graph, the Soviets were first to achieve Satellite and Suborbital, but the Americans then achieved Orbital and Spacewalk.

![Ranking Comparison](https://user-images.githubusercontent.com/3118014/193423154-455d4326-ba65-4a4a-b608-dffa17e7ab73.png)

**Reply by hweimer** [reply]
@peyre:
> How do we enable PBEM, by the way?

Run cmake with "-DPBEM=ON"

**Reply by peyre** [reply]
Thanks!

Edit:  Funny, I ran it with that option and the game compiled, but Play By Email didn't appear in the main menu and pressing P doesn't open a PBEM game.  This is the command I ran to compile the game:

`cd ~/ && sudo rm -r raceintospace-build && mkdir raceintospace-build && cd raceintospace-build && cmake -DBPEM=ON -DCMAKE_BUILD_TYPE=Debug ../raceintospace && make && sudo make install && cd src/game && ./raceintospace`

(This, of course, was my own fault, a 10T error.  It's not going to work if you enter -DBPEM instead of -DPBEM ! )

**Reply by hweimer** [reply]
Actually, the "pivotal moment" thing might be due to a bug. I just discovered that in most situations, prestige awarded for missions (both positive and negative) was counted twice, see #694. This was not very noticeable because the prestige summary in the admin building showed the correct score. But the part where it matters the most for the game, i.e., the budget, used the wrong values. So you could get a much larger boost from early successes than how it should be.

**Reply by peyre** [reply]
No kidding!  So that might reduce the impact of this sort of thing.  Must start testing it this weekend - maybe even tomorrow, since I have the day off.

**Comment by peyre** [comment]
Another thought is that it might be nice if the newscast made some effort to spread out hits to your R&D, so you won't have a series of setbacks just getting something up and running, like:

![1](https://user-images.githubusercontent.com/3118014/194770376-a4c2a50f-5c59-46d2-940c-4629c67f9a65.png)
![2](https://user-images.githubusercontent.com/3118014/194770382-e12c0871-3e6f-4d0d-8673-35aca9df1d7b.png)
![3](https://user-images.githubusercontent.com/3118014/194770386-12c01c4e-f9bf-43e5-9153-e3a80419d0e3.png)

Especially when it's immediately followed by a rocket _above_ Max R&D exploding on the pad:
![Exploded](https://user-images.githubusercontent.com/3118014/194770488-cba0e83b-b613-4453-a4a6-e404b529b4d9.png)

And then my capsule (also above Max R&D) burning up on reentry.  It's Fall 1963 and I can't even fly a manned mission without killing whoever's aboard.
![Reentry](https://user-images.githubusercontent.com/3118014/194770742-46be1573-1bfe-4342-aa5b-a83fce563c7b.png)

So I take the hint that Vostok's just not going to work in this game, and build up my Soyuz, manage to do a suborbital, but get swatted for trying anything more ambitious.  It's Spring 1966 and my people can't spend a few days in space without paying for it with their lives.  When I did get the program up and running, I managed to fly a duration C just before the computer won the game with Apollo III.

**Comment by peyre** [comment]
On the flip side of course, it would be good to find some way of counterbalancing when someone's getting all good rolls, especially if his opponent is having a bad time.  In this last game, I played the US and it seemed I could do no wrong, and the AI just couldn't get ahead.

![Unbalanced2](https://user-images.githubusercontent.com/3118014/209576972-666a03d7-c51b-475c-92e3-e888e50783f9.png)

**Comment by peyre** [comment]
Aaand, another one.  EVERY SINGLE TIME I try to fly a manned mission, it fails - usually catastrophically.  Mercury gets trashed so I abandon it in favor of Apollo.  I watch the AI get two, three prestige firsts per turn while I'm struggling with research.

Finally in **Fall 1965** I'm ready to fly my first g--d--ned Suborbital mission and my Max R&D capsule burns like a wad of tissue paper.  Then our friendly newscaster tells me I haven't had enough misfortune yet, so I _also_ get to pay extra for more Apollos.  "That's a nice space program you have there.   Be a shame if something happened to it.  --Oh, it did!"

![Unfair1](https://user-images.githubusercontent.com/3118014/224508303-327783a4-6fdc-4512-b683-a055c42cf0b1.png)
![Unfair2](https://user-images.githubusercontent.com/3118014/224508302-f8f58062-c988-4a05-8a2f-904fc77b0034.png)

**Comment by peyre** [comment]
I just had a thought about something that might reduce this issue.  What if we did something like (maybe after your first catastrophic failure), further cat fails--as long as the hardware was at or above Max R&D--cost you half as much prestige, to reflect that you're at least doing your due diligence as regards safety?  

Or something more graduated, such as reducing the penalty by 40% if you're 1 or 2 points from Max R&D, 50% if you're at Max, or 60% if you're 2 or more points above Max?

**Comment by peyre** [comment]
Another possibility occurs.  One of the most frustrating parts of this sort of thing is that multiple cat failures tank your budget, which makes it almost impossible to catch up to your opponent.  How about if the usual hit to your budget occurs only if yours is currently higher than your opponent's?  

Maybe your government doesn't vindictively slash your budget when it can see that it's already spending less on the project than the competition.

**Comment by peyre** [comment]
Ah, here - early in the game.  How am I supposed to recover from something like this, especially as the Soviets?  I mean it's not like I was taking huge risks, skipping milestones and flying things under Max R&D.  I'd even planned to fly a Vostok dummy test, but the newscast decided to block that by cutting off rockets when I needed them.  My first satellite failed and the Americans slipped in, then they fly an orbital EVA before I can even attempt a suborbital, and that fails - so my first orbital has to downgrade to a suborbital, and the game slaps that down hard too.

![GraphDisparity](https://github.com/user-attachments/assets/885d8985-76ba-4101-9fd9-d93c1a1b3868)

**Comment by hweimer** [comment]
Sort of related: Wasn't there a change that led dummy tests before Max R&D to also increase Max R&D by one point? Seems like this is no longer the case. Is this intentional?

**Reply by peyre** [reply]
Yes, Nicolas Laplume did that in v. 1.0.  It still does it, pretty sure.

It makes sense for the game to do it, really.  Otherwise a dummy test below Max R&D is wasted.

**Reply by hweimer** [reply]
I think it's even worse. Even at Max R&D, the safety factor is not increased by a successful step.

**Reply by peyre** [reply]
Uh oh.  Maybe something we did recently broke that?

**Comment by peyre** [comment]
Here we are again.  I'm just 5 turns in, and everything's working against me.  Three satellite launches fail in succession, the last destroying my pad.  Meanwhile the newscasts keep degrading my equipment.

<img width="650" height="424" alt="1" src="https://github.com/user-attachments/assets/19fb7840-fbf8-4073-ae93-2a36c305a79f" />
<img width="650" height="424" alt="2" src="https://github.com/user-attachments/assets/f2ebad9a-3bf9-4acc-8369-8954e1326f6b" />
<img width="650" height="424" alt="3" src="https://github.com/user-attachments/assets/7ba76760-9148-4fd7-9bfc-30f504c506fb" />
<img width="650" height="424" alt="4" src="https://github.com/user-attachments/assets/6c2ff6aa-0c75-4894-ac80-650798735227" />
<img width="650" height="424" alt="5" src="https://github.com/user-attachments/assets/72c14afd-a6d7-4883-bc11-6a90a8a7f439" />


Ok, one newscast improved my hardware rather than downgrading it.  Still...

<img width="650" height="424" alt="6" src="https://github.com/user-attachments/assets/e0fa1c2a-d19b-47dd-a104-ba6ec286f9d3" />



Then, of course, once I finally do manage to launch a satellite successfully, my first manned suborbital (at Max R&D) is a catastrophic failure.

<img width="650" height="424" alt="7" src="https://github.com/user-attachments/assets/8a428920-5483-4a14-8735-af43a1e25807" />

**Comment by peyre** [comment]
This is from the other end.  I'm playing as the Soviets, and apparently I can do no wrong, while my opponent is getting slapped around.  I've gotten every prestige first and the AI, multiple catastrophic failures apparently.  I know that's just how random rolls work, but the game really needs some kind of mitigation for this level of unevenness.  If only we could think of one that was workable.

<img width="650" height="424" alt="Uneven" src="https://github.com/user-attachments/assets/df07aae5-8176-4d04-acf0-17b94415fe20" />

**Reply by Angel-0a** [reply]
Maybe increase the number of "Avoid failure" cards and allow players to hold more than one? This would give players a fighting chance against bad luck.

**Reply by peyre** [reply]
Well there's an idea!  I was thinking, possibly two Avoid Fails each to the capsule and rocket they last used.  But then I realized that might be creating an exploit.  Of course, granting them only after the downgraded item returns to Max R&D would reduce that.

**Reply by Angel-0a** [reply]
Only after posting my idea I realized it was actually the first idea proposed in this discussion, including some very interesting variations that take dummy tests and R&D into account.

**Comment by peyre** [comment]
I just had another pondering.  What if the chances of a failure going catastrophic were to decrease when you're more than, say, 30 prestige points behind your opponent?  Then again we'd have to be careful with something like that, because it could play havoc with the realism, and I'm more enthusiastic about having the relevant crew member's skill affect your chance of a failure becoming catastrophic, especially for the specialty skills (LM, EV, DO).

It might still be a better idea to simply reduce the effect of prestige loss on your budget when you're already behind.  Like, in the example above, I'm about 70 points ahead of the Americans.  I believe the budget is raised/lowered by the same number of MBs as the prestige gain/loss?  Maybe on a cat failure, the hit to your budget is reduced by 1 for every 10 points you're behind your opponent.  So if the Americans had a full crew death next turn (-10 prestige), their budget would go down by 3 instead of 10.

**Reply by Angel-0a** [reply]
> What if the chances of a failure going catastrophic were to decrease when you're more than, say, 30 prestige points behind your opponent?

This one I don't like. It sounds like rubberbanding. Anyway - you should push for implementation of some of these ideas in a test branch and start play testing them. You don’t have to wait for the perfect idea. More Avoid Fail cards may be a good start as this one will probably be the easiest one (although not necessarily easy) to implement. The underlying code is already in the game I guess.

**Comment by peyre** [comment]
Also, maybe on a failure, the chance of it going catastrophic would be, say, +5% if you're at or above Max R&D, -5% if you're below.
